### PR TITLE
Kestrel + fake HttpClient with HttpRequest, HttpRequestMessage

### PR DESCRIPTION
**Pull request by alnikola**
…esponseMessage, HttpResponse

### CODE REVIEW

**Comment by alnikola** [comment]
@Tratcher I used [this existing header copying code](https://github.com/aspnet/Benchmarks/blob/main/src/BenchmarksApps/HttpClient/Proxy/ProxyExtensions.cs) from HttpClient proxy as the template. Do you mean that this code is incorrect too?

**Comment by alnikola** [comment]
Ah, got it. There is already ContentLength set by the downstream. I fix my test then.

File: src/BenchmarksApps/HttpClient/Proxy/ProxyExtensions.cs [hunk 1]
```diff
@@ -126,6 +127,14 @@ public static async Task CopyProxyHttpResponse(this HttpContext context, HttpRes
             }
         }
 
+        public static HttpResponseMessage CreateResponse(HttpRequestMessage request, byte[] responseBody)
+        {
+            var response = new HttpResponseMessage(System.Net.HttpStatusCode.OK);
+            CopyHeaders(response.Headers, request.Headers);
+            response.Content = new ByteArrayContent(responseBody);
```
Comment (Tratcher) [inline src/BenchmarksApps/HttpClient/Proxy/ProxyExtensions.cs:134]: You still need to set the ContentLength.

Comment (alnikola) [reply src/BenchmarksApps/HttpClient/Proxy/ProxyExtensions.cs:134]: Done, but I don't see any impact on RPS.

Comment (alnikola) [reply src/BenchmarksApps/HttpClient/Proxy/ProxyExtensions.cs:134]: I discovered that it does increase RPS but only in one scenario - 1024B.

**Pull request closed without merging**